### PR TITLE
[FSPE-7168] Added link component

### PR DESCRIPTION
**Pull request by mattnickles**
https://ctl.atlassian.net/browse/FSPE-7168

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://3b2913ee419e6b3b5f36cf8db8f1609f6523f0d9-chi-pr.enterprise.ctl.io" target="_blank"> https://3b2913ee419e6b3b5f36cf8db8f1609f6523f0d9-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1774/ for more details.

**Review by jllr — APPROVED** [review]

File: src/website/views/components/link/_examples.pug [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        <div class="chi-link__content">
+          <i class="chi-icon icon-circle-question-outline"></i>
+          <span>Help</span>
+        </div>
+      </a>
+
+h3 Icon Link
+p.-text
+  | Use icon links to render icons inside link tags.
+  | <strong> Note:</strong> All icon links without text lables
```
Comment (SergioQCL) [inline src/website/views/components/link/_examples.pug:92]:
```suggestion
  | <strong> Note:</strong> All icon links without text labels
```

File: src/custom-elements/src/components/link/link.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   *  to set the relationship between the current page and the linked page.
+   */
+  @Prop({ reflect: true }) rel: string;
+
+  /**
+   *  to set size of a link { xs, sm, md, lg, xl }.
+   */
+  @Prop({ reflect: true }) size: string;
```
Comment (SergioQCL) [inline src/custom-elements/src/components/link/link.tsx:51]: Do we need the 'md' size, since it's the same as not setting a size?
Also, maybe it's a good idea to create a sizes' set type and defaulting this to 'md' in case we want to keep that value. Thoughts?

File: src/custom-elements/src/components/link/link.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   *  to set target of a link { self, blank, parent, top }.
+   */
+  @Prop({ reflect: true }) target: string;
+
+  @Watch('size')
+  sizeValidation(newValue: string) {
+    if (newValue && !['', 'xs', 'sm', 'md', 'lg', 'xl'].includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for link. Valid values are xs, sm, md, lg, xl, ''. `);
```
Comment (SergioQCL) [inline src/custom-elements/src/components/link/link.tsx:61]: I think including an empty string in the comparison is not necessary, since _newValue_ would return false. Because of that, I would reword the Error like this:
```suggestion
    if (!!newValue && !['', 'xs', 'sm', 'md', 'lg', 'xl'].includes(newValue)) {
      throw new Error(`${newValue} is not a valid size for link. If provided, valid values are xs, sm, md, lg, or xl. `);
```

File: src/custom-elements/src/components/link/link.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  }
+
+  @Watch('target')
+  targetValidation(newValue: string) {
+    if (newValue && !['', '_self', '_blank', '_parent', '_top'].includes(newValue)) {
+      throw new Error(`${newValue} is not a valid target for a link. Valid values are _self, _blank, _parent, _top, or ''. `);
```
Comment (SergioQCL) [inline src/custom-elements/src/components/link/link.tsx:68]: Same case here

**Review by SergioQCL — CHANGES_REQUESTED** [review]

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://46056fbf3efc527347f5d25c7d351f9a1f7f75d3-chi-pr.enterprise.ctl.io" target="_blank"> https://46056fbf3efc527347f5d25c7d351f9a1f7f75d3-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1776/ for more details.

File: src/custom-elements/src/components/link/link.tsx (hunk 2 above)
Comment (mattnickles) [reply src/custom-elements/src/components/link/link.tsx:51]: @SergioQCL Links behave similar to text in that they inherit the font-size of the parent by default. This would allow a scenario where the default size and md size are not the same. For this reason I would suggest not defaulting to md.

File: src/custom-elements/src/components/link/link.tsx (hunk 4 above)
Comment (mattnickles) [reply src/custom-elements/src/components/link/link.tsx:68]: Thanks for catching this - this should actually be removed as target supports custom named frames.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1777/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1778/ for more details.

**Review by SergioQCL — APPROVED** [review]
LGTM

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1779/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://f9a44ca6bd15d310749ef6fac621fc4efad9eabb-chi-pr.enterprise.ctl.io" target="_blank"> https://f9a44ca6bd15d310749ef6fac621fc4efad9eabb-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1780/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-7wqvg0, https://ops.elasticbox.com/services/instances/i-h2vje6, https://ops.elasticbox.com/services/instances/i-v6rgsx
